controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -118,15 +120,24 @@ public ControllerService(StreamMetadataStore streamStore, StreamMetadataTasks st
                     CreateStreamStatus.newBuilder().setStatus(CreateStreamStatus.Status.INVALID_STREAM_NAME).build());
         }
 
-        return streamMetadataTasks.createStream(scope,
-                                                stream,
-                                                streamConfig,
-                                                createTimestamp)
-                  .thenApplyAsync(status -> {
-                       reportCreateStreamMetrics(scope, stream, streamConfig.getScalingPolicy().getMinNumSegments(), status,
-                                timer.getElapsed());
-                       return CreateStreamStatus.newBuilder().setStatus(status).build();
-                  }, executor);
+        return Futures.exceptionallyExpecting(streamStore.getState(scope, stream, true, null, executor), 
+            e -> Exceptions.unwrap(e) instanceof StoreException.DataNotFoundException, State.UNKNOWN) 
+        .thenCompose(state -> {
+            if (state.equals(State.UNKNOWN) || state.equals(State.CREATING)) {
```

Comment:
I don't understand why we need to try to create again in the case that the stream is `State.CREATING`? Doesn't it mean that it is in the process of being created?

Reply:
A previous attempt to create a stream could have failed and left the stream in CREATING state.